queue_job_batch/models/queue_job_batch.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models, fields, api
+from odoo.addons.queue_job.job import job
+
+
+class QueueJobBatch(models.Model):
+    _name = 'queue.job.batch'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = 'Batch of jobs'
+    _log_access = False
+
+    name = fields.Char(
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    job_ids = fields.One2many(
+        'queue.job',
+        inverse_name='job_batch_id',
+        readonly=True,
+    )
+    job_count = fields.Integer(
+        compute='_compute_job_count',
+    )
+    user_id = fields.Many2one(
+        'res.users',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('progress', 'In Progress'),
+            ('finished', 'Finished')
+        ],
+        default='draft',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    finished_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        readonly=True,
+    )
+    is_read = fields.Boolean(
+        default=True
+    )
+    completeness = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_percentage = fields.Float(
+        compute='_compute_job_count',
+    )
+
+    @job
+    def check_state(self):
+        self.ensure_one()
+        if self.state == 'draft' and any(
+            job.state not in ['pending', 'enqueued'] for job in self.job_ids
+        ):
+            self.write({'state': 'progress'})
```

Comment:
Here there might be a race condition in this scenario:
1. create batch and jobs in one transaction
2. create more jobs in the same batch in another transaction

If between 1 and 2 all jobs finish, the batch will go to done state before jobs created in 2 are seen.

So maybe batch.enqueue() is necessary after all.

Reply:
But the batch.enqueue has no sense without the pending state... :thinking: 
Maybe we could add the state on `queue_job` module in order to be reused in other modules?

Reply:
I mean you could have the following states for the batch:
1. jobs being added (draft)
2. all jobs created (this state is set when the developer signals that all jobs are created, I thought it was the purposed of `enqueued`, but maybe another name is better)
3. progress at least one job not in (pending, enqueued)
4. done all jobs done

And a draft batch remains batch even if all jobs are done.

Reply:
Done! :smile: